### PR TITLE
doc: consolidate library documentation to libraries.md

### PR DESCRIPTION
**Pull request by fanquake**
Deduplicate the makefile comments, in favour of doc/libraries.md. I think a single, more comprehensive source of truth is preferable. Diagrams are also useful. Came up in https://github.com/bitcoin/bitcoin/pull/26292#issuecomment-1275094478.

### CODE REVIEW

**Review by hebasto — COMMENTED** [review]
> Deduplicate the makefile comments...

Concept ACK on that (especially, in the light of #25797).

File: src/Makefile.am [hunk 1]
```diff
@@ -439,7 +435,9 @@ endif
 if !ENABLE_WALLET
 libbitcoin_node_a_SOURCES += dummywallet.cpp
 endif
+#
```
Comment (yancyribbens) [inline src/Makefile.am:438]: @fanquake, just curious why the trailing comment line here without any comment.  Is this for automated documentation?

**Review by ryanofsky — APPROVED** [review]
Code review ACK af781bf4b2998eb17e89b6b24d26a2590e548259, nice cleanups

**Review by hebasto — APPROVED** [review]
ACK af781bf4b2998eb17e89b6b24d26a2590e548259, I have reviewed the code and it looks OK, I agree it can be merged.

**Pull request merged**